packages/core/src/browser/shell/tab-bar-toolbar.tsx

```diff
@@ -65,7 +65,7 @@ export class TabBarToolbar extends ReactWidget {
 
     protected render(): React.ReactNode {
         return <React.Fragment>
-            {[...this.items.values()].map(item => this.renderItem(item))}
+            {[...this.items.values()].map(item => TabBarToolbarItem.is(item) ? this.renderItem(item) : item.node)}
```

Comment:
Not directly in the scope of this PR, but what is the difference between returning a `<React.Fragment>` and just returning the mapped list directly? I tried and did not notice a difference.

Reply:
A [`Fragment` IS_A `ReactNode`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/7c77a9c53a5d3e3c3c0a29a466b5acf810cac7a4/types/react/index.d.ts#L194).

Reply:
Hm, I was more wondering why we don't do the following:
```ts
    protected render(): React.ReactNode {
        return [...this.items.values()].map(item =>
            TabBarToolbarItem.is(item) ? this.renderItem(item) : item.render());
    }
```

Reply:
I do not know; if you feel uncomfortable with the current code, please file an issue.